### ISSUE TITLE
Charts are blank

### ISSUE THREAD

**Issue by ISO8807** [wp-statistics/wp-statistics#410]
For about a year now, WP-Statistics has continued to collect data. However, all charts are blank. I have the latest version of WP (6.0.1) and WP-Statistics (13.2.5).

Each chart is pale blue with a light band sweeping from left to right. As an example, see:

![wp-statistics-charts](https://user-images.githubusercontent.com/40432029/184861573-3b61e129-d3ce-4300-a956-3ed0b67d6fba.jpg)

The "bug" is that this is failing without any indication of what is wrong. Turning on WP debugging yields no useful information (there are no PHP errors). Perhaps the documentation does not specify the requirements fully.

**Comment by mostafasoufi** [comment]
Hi,

Thank you for reporting the issue, can you please tell me the status and response of the requests in **Inspect > Network > XHR**?

![[MarineGEO circle logo](https://wp-statistics.com/wp-content/uploads/2021/03/Screen-Shot-2021-03-13-at-12.34.26-PM.png)](https://wp-statistics.com/wp-content/uploads/2021/03/Screen-Shot-2021-03-13-at-12.34.26-PM.png)

**Comment by ISO8807** [comment]
Many thanks for the response. I attach a typical pair of messages as a text (JSON) file after trying to show hits. This happens to be on my local PC (Windows 10), but I get the same behaviour on a networked system. I recorded the XHR using Firefox, which seems to have an equivalent feature to what you asked about.

I have a suspicion that something like my use of the WP-Hide plugin may be causing the problem as it rewrites a lot of URL.s. Certainly .../wp-json is not a valid URL.

Following a 301 redirection there's a 404 response to a GET for "...//localhost/wp-json/wp-statistics/v2/metabox/?ago=30&name=hits&_=1660735872975" in the following:

[wp-statistics-xhr.txt](https://github.com/wp-statistics/wp-statistics/files/9359165/wp-statistics-xhr.txt)

**Comment by ISO8807** [comment]
Your question was enough for me to fix this.

Indeed, WP-Hide was the cause of the "/wp-json" REST call failing. I had disabled this for security reasons, though the consequences were not obvious. It appears that the critical WP-Hide settings in Rewrite/URLs > JSON REST should be:

  Disable JSON REST V2 service = No
  
  Block any JSON REST calls = Non logged-in
  
This leaves an insecurity in that the site can be probed for "/wp-jso" URLs. I worked around this by adding a filter in the theme's functions.php for "rest_url_prefix". I also had to manually add a rule to .htaccess to rewrite this prefix to "/wp-json" (which is directly supported in the Pro version of WP-Hide).

Unless I missed it, it would be useful if the WP-Statistics documentation noted the need for the JSON REST service to be enabled.

Thanks for your time.
